### PR TITLE
Handle missing log collector for sidecar configurations.

### PR DESCRIPTION
**Pull request by dennisoelkers**
## Motivation and Context

This PR is fixing an issue when a sidecar configuration is referencing a log collector which is not present anymore. In this case, the Sidecar/Configuration page throws an exception and is not usable anymore.

It is currently unclear which steps can lead to a condition like this, because deleting a log collector from the frontend is not possible when it is referenced, but it was seen in the wild.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by maxiadlovskii — APPROVED** [review]

File: graylog2-web-interface/src/components/sidecars/common/SidecarSearchForm.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+          <tr>
+            <td>sidecar_version</td>
+            <td>Sidecar version</td>
+          </tr>
+        </tbody>
+      </Table>
+      <p><strong>Examples</strong></p>
+      <p>
+        Find sidecars that did not communicate with Graylog since a date:<br />
+        <kbd>{'last_seen:<=2018-04-10'}</kbd><br />
```
Comment (linuspahl) [inline graylog2-web-interface/src/components/sidecars/common/SidecarSearchForm.tsx:73]: Let's remove `{''}` here.

File: graylog2-web-interface/src/components/sidecars/common/SidecarSearchForm.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        <kbd>status:failing status:unknown</kbd><br />
+      </p>
+    </Popover>
+  );
+
+  const queryHelp = (
+    <OverlayTrigger trigger="click" rootClose placement="right" overlay={queryHelpPopover}>
+      <Button bsStyle="link"><Icon name="question-circle" /></Button>
+    </OverlayTrigger>
+  );
```
Comment (linuspahl) [inline graylog2-web-interface/src/components/sidecars/common/SidecarSearchForm.tsx:86]: Looks like we can define `queryHelpPopover` and  `queryHelp` outside of the component.

**Review by linuspahl — COMMENTED** [review]
Just found some smaller things we can improve, feel free to merge the PR already.

**Comment by dennisoelkers** [comment]
Superseded by #21020.

**Pull request closed without merging**